File: packages/javascript/lib/string/capitalizeEverySentence.ts
const ASCII_LETTER = /[a-zA-Z]/;

export function capitalizeEverySentence(str: string, splitChar?: string): string {
	if (!str) {
		return '';
	}

	const splitter: string = splitChar || '.';
	const splitStr = str.split(splitter);
	let resultStr = '';
	let sentenceChars;

	for (let i = 0, iLen = splitStr.length; i < iLen; i += 1) {
		// Iterate by code point. Indexing the raw string here would drift out of sync
		// with this array as soon as the sentence contains a surrogate pair (emoji).
		sentenceChars = [...splitStr[i]];

		for (let j = 0, jLen = sentenceChars.length; j < jLen; j += 1) {
			if (ASCII_LETTER.test(sentenceChars[j])) {
				sentenceChars[j] = sentenceChars[j].toUpperCase();
				break;
			}
		}

		resultStr += `${sentenceChars.join('')}${i < iLen - 1 ? splitter : ''}`;
	}

	return resultStr;
}

File: packages/python/qsu/string/capitalizeEverySentence.py
import re


def capitalizeEverySentence(str: str, splitChar: str = None) -> str:
	if not str:
		return ''

	splitter = splitChar or '.'
	splitStr = str.split(splitter)
	resultStr = ''
	iLen = len(splitStr)

	for i in range(iLen):
		sentenceChars = list(splitStr[i])

		for j in range(len(sentenceChars)):
			if re.match(r'[a-zA-Z]', splitStr[i][j]):
				sentenceChars[j] = splitStr[i][j].upper()
				break

		resultStr += ''.join(sentenceChars) + (splitter if i < iLen - 1 else '')

	return resultStr
